### PR TITLE
fixed few minor issues (#39)

### PR DESCRIPTION
**Pull request by monkeyman192**
Few parameters had incorrect default values. Sidecar file now can have task and run info included.

### CODE REVIEW

**Comment by monkeyman192** [comment]
Huh, so apparently the test fail because it thinks the value of the 'SoftwareFilters' parameter should be "n/a", but the specification clearly states that "none" is to be used if no filters are used... (cf. pg6 of https://docs.google.com/document/d/1FWex_kSPWVh_f4rKgd5rxJmxlboAPtQlmBc1gyZlRZM/edit)

**Comment by agramfort** [comment]
cc @jasmainak @robertoostenveld

**Comment by jasmainak** [comment]
You are looking at the wrong document. The BEP008 draft is no longer the right document since it has been merged into the main specification. See here: https://docs.google.com/document/d/1HFUkAEE-pB-angVcYe6pf_-fVf4sCpOHKesUvfb8Grc/edit#. It clearly states you should use "n/a" not "none" as in the rest of BIDS

**Comment by robertoostenveld** [comment]
Please don't use the old work-in-progress version of the BEP008 MEG extension, but the final merged version at https://docs.google.com/document/d/1HFUkAEE-pB-angVcYe6pf_-fVf4sCpOHKesUvfb8Grc. The specifications in the two versions are not 100% identical. 

The 1.1.0 spec specifies "Write “n/a” if no software filters applied".

**Comment by robertoostenveld** [comment]
@jasmainak beat me by 15 seconds...

**Comment by jasmainak** [comment]
we live in a fast paced world :)

**Comment by monkeyman192** [comment]
Oh, my bad!
I thought I had the most recent specifications!!
It only looks like the software filters value affects my PR, so I guess I'll revert that and then keep the PR with the other two changes?

**Comment by jasmainak** [comment]
sounds reasonable to me

**Comment by monkeyman192** [comment]
Ok, I tried to make a few more changes.
I added a new class to deal with file naming instead of using the current method.
I think this would allow the naming process to be a lot more robust, and far more easily called in the raw_to_bids function. The folder creation could also be incorporated into this class.
It still needs a little bit of work, but I wanted to see whether people thought it would be worth using?
I am not exactly sure why the test is failing.
One of the errors seems to not like the .sqd file in the same directory as the other files, and says that the souce files should be in a ./sourcedata/ folder, however the MEG specification states to put it in the same folder as the _meg.json file?

File: mne_bids/meg_bids.py [hunk 1]
```diff
@@ -305,25 +318,14 @@ def raw_to_bids(subject_id, task, raw_file, output_path, session_id=None,
                                      verbose=verbose)
 
     # create filenames
-    scans_fname = make_bids_filename(
-        subject=subject_id, session=session_id, suffix='scans.tsv',
-        prefix=ses_path)
-
-    coordsystem_fname = make_bids_filename(
```
Comment (jasmainak) [inline mne_bids/meg_bids.py]: I'm not a big fan of using an object here, but maybe it's just me. @agramfort any opinions?

**Comment by jasmainak** [comment]
> One of the errors seems to not like the .sqd file in the same directory as the other files, and says that the souce files should be in a ./sourcedata/ folder, however the MEG specification states to put it in the same folder as the _meg.json file?

Okay, this is my bad. The validator would need to be updated I guess

**Pull request closed without merging**